unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -165,6 +165,20 @@ public UTF8String toLowerCase() {
     return UTF8String.fromString(toString().toLowerCase());
   }
 
+  public UTF8String initCap(final byte[] byteArr) {
+    if (byteArr == null) return null;
+    if (byteArr[0] <= 122 && byteArr[0] >= 97) //a-z
```

Comment:
I don't agree with that.

Apache commons lang wordutils can handle this:
http://grepcode.com/file/repo1.maven.org/maven2/commons-lang/commons-lang/2.6/org/apache/commons/lang/WordUtils.java#WordUtils.capitalize%28java.lang.String%29

```
scala> Character.toTitleCase('ä')
res52: Char = Ä
```

There seems to be not `toUpper` of Java `StringBuilder` 
http://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html

`String` does it support:
```
scala> "äöü".toUpperCase()
res57: String = ÄÖÜ
```